java-to-zod-core/src/main/java/sh/ivan/zod/plugins/PluginParameters.java

```diff
@@ -0,0 +1,320 @@
+     * <li><code>jaxb</code> - annotations from `javax.xml.bind.annotation` package<li>
+     * <li><code>gson</code> - annotations from `com.google.gson.annotations` package<li>
+     * <li><code>jsonb</code> - annotations from `javax.json.bind.annotation` package<li>
+     * </ul>
+     * Required parameter, recommended value is <code>jackson2</code>.
+     */
+    @Parameter(required = true)
+    private JsonLibrary jsonLibrary;
+
+    /**
+     * Specifies Jackson 2 global configuration.
+     * Description of individual parameters is in
+     * <a href="https://github.com/vojtechhabarta/typescript-generator/blob/main/typescript-generator-core/src/main/java/cz/habarta/typescript/generator/Jackson2Configuration.java">Jackson2Configuration</a>
+     * class on GitHub (latest version).
+     */
+    @Parameter
+    private Jackson2Configuration jackson2Configuration;
+
+    /**
+     * Specifies Gson global configuration.
+     * Description of individual parameters is in
+     * <a href="https://github.com/vojtechhabarta/typescript-generator/blob/main/typescript-generator-core/src/main/java/cz/habarta/typescript/generator/GsonConfiguration.java">GsonConfiguration</a>
+     * class on GitHub (latest version).
+     */
+    @Parameter
+    private GsonConfiguration gsonConfiguration;
+
+    /**
+     * Specifies JSON-B global configuration.
+     * Description of individual parameters is in
+     * <a href="https://github.com/vojtechhabarta/typescript-generator/blob/main/typescript-generator-core/src/main/java/cz/habarta/typescript/generator/JsonbConfiguration.java">Jackson2Configuration</a>
+     * class on GitHub (latest version).
+     */
+    @Parameter
+    private JsonbConfiguration jsonbConfiguration;
+
+    /**
+     * If <code>true</code> Spring REST application will be loaded and scanned for classes to process.
+     * It is needed to specify application class using another parameter (for example {@link #classes}).
+     */
+    @Parameter
+    private boolean scanSpringApplication = false;
+
+    /**
+     * Specifies level of logging output.
+     * Supported values are:
+     * <ul>
+     * <li><code>Debug</code></li>
+     * <li><code>Verbose</code></li>
+     * <li><code>Info</code></li>
+     * <li><code>Warning</code></li>
+     * <li><code>Error</code></li>
+     * </ul>
+     * Default value is <code>Verbose</code>.
+     */
+    @Parameter
+    private Logger.Level loggingLevel = Logger.Level.Verbose;
+
+    @Parameter(property = "java.to.zod.skip")
+    private boolean skip = false;
+
+    /**
+     * The presence of any annotation in this list on a JSON property will cause
+     * the typescript-generator to treat that property as optional when generating
+     * the corresponding TypeScript interface.
+     * Example optional annotation: <code>javax.annotation.Nullable</code>.
+     */
+    private List<String> optionalAnnotations = Collections.emptyList();
+
+    @Input
+    @Optional
```

Comment:
For Optional Inputs, we can use a variation of Getter on the specific field declaration:
```java
@Getter(onMethod=@__({@Input, @Optional}))
private String schemaNameSuffix = "Schema";
```